src/renderer/components/dock/log-tab.store.ts

```diff
@@ -0,0 +1,100 @@
+import uniqueId from "lodash/uniqueId";
+import { podsStore } from "../+workloads-pods/pods.store";
+
+import { IPodContainer, Pod } from "../../api/endpoints";
+import { WorkloadKubeObject } from "../../api/workload-kube-object";
+import { DockTabStore } from "./dock-tab.store";
+import { dockStore, IDockTab, TabId, TabKind } from "./dock.store";
+
+export interface LogTabData {
+  pods: Pod[];
+  selectedPod: Pod;
+  selectedContainer: IPodContainer
+  containers?: IPodContainer[]
+  initContainers?: IPodContainer[]
+  showTimestamps?: boolean
+  previous?: boolean
+}
+
+interface PodLogsTabData {
+  selectedPod: Pod
+  selectedContainer: IPodContainer
+}
+
+interface WorkloadLogsTabData {
+  workload: WorkloadKubeObject
+}
+
+export class LogTabStore extends DockTabStore<LogTabData> {
+  constructor() {
+    super({
+      storageName: "pod_logs"
+    });
+  }
+
+  public createPodTab({ selectedPod, selectedContainer }: PodLogsTabData) {
```

Comment:
I don't think it's necessary. This `void`type is determined automatically by typescript so its visible in IDE. And I'm trying currently to stick with clear names, meaning if function is called like `createPodTab` then it shouldn't return anything. Otherwise it would be named like `getPodTab()` or similar. But I can add `:void` notation by hand if you want 😃

Reply:
I guess I view it as part of the function contract. Namely, changing the return value is generally a breaking change so I am of the opinion that they should be always explicitly marked (except for functions like `render**` or maybe `toString`)